goaviatrix/remote_syslog.go

```diff
@@ -0,0 +1,121 @@
+package goaviatrix
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"strconv"
+)
+
+type RemoteSyslog struct {
+	CID                 string   `form:"CID,omitempty"`
+	Server              string   `form:"server,omitempty" json:"server"`
+	Port                int      `form:"port,omitempty" json:"port"`
+	Protocol            string   `form:"protocol,omitempty" json:"protocol"`
+	Index               int      `form:"index,omitempty" json:"index"`
+	Template            string   `form:"template,omitempty" json:"template"`
+	CaCertificate       string   `form:"ca_certificate,omitempty"`
+	PublicCertificate   string   `form:"public_certificate,omitempty"`
+	PrivateKey          string   `form:"private_key,omitempty"`
+	ExcludeGatewayInput string   `form:"exclude_gateway_list,omitempty"`
+	ExcludedGateway     []string `json:"excluded_gateway"`
+	Status              string   `json:"status"`
+	Notls               bool     `json:"notls"`
+}
+
+type RemoteSyslogResp struct {
+	Server          string   `json:"server"`
+	Port            string   `json:"port"`
+	Protocol        string   `json:"protocol"`
+	Index           string   `json:"index"`
+	Template        string   `json:"template"`
+	ExcludedGateway []string `json:"excluded_gateway"`
+	Status          string   `json:"status"`
+	Notls           bool     `json:"notls"`
+}
+
+func (c *Client) EnableRemoteSyslog(r *RemoteSyslog) error {
+	params := map[string]string{
+		"action":   "enable_remote_syslog_logging",
+		"CID":      c.CID,
+		"index":    strconv.Itoa(r.Index),
+		"server":   r.Server,
+		"port":     strconv.Itoa(r.Port),
+		"protocol": r.Protocol,
+		"template": r.Template,
+	}
+
+	files := []File{
+		{
+			Path:      r.CaCertificate,
+			ParamName: "ca_certificate",
+		},
+		{
+			Path:      r.PublicCertificate,
+			ParamName: "public_certificate",
+		},
+		{
+			Path:      r.PrivateKey,
+			ParamName: "private_key",
+		},
+	}
+
+	resp, err := c.PostFile(c.baseURL, params, files)
```

Comment:
Check out `PostFileAPI` function, it is a helper to avoid rewriting the response decoding and error checking